src/arch/host/include/arch/sof.h

```diff
@@ -10,7 +10,9 @@
 #ifndef __ARCH_SOF_H__
 #define __ARCH_SOF_H__
 
+#if !defined(__ZEPHYR__) || !CONFIG_LIBRARY
```

Comment:
I think it's probably best we create a zephyr/include/arch/sof.h wrapper - this can be intermediate and directly call the zephyr equivalents.